app/src/main/java/ch/epfl/balelecbud/EmergencyNumbersActivity.java

```diff
@@ -0,0 +1,124 @@
+package ch.epfl.balelecbud;
+
+import android.Manifest;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.net.Uri;
+import android.os.Bundle;
+import android.util.Log;
+import android.widget.ArrayAdapter;
+import android.widget.ListView;
+
+import androidx.core.app.ActivityCompat;
+
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import ch.epfl.balelecbud.emergency.models.EmergencyNumber;
+import ch.epfl.balelecbud.models.emergency.Emergency;
+import ch.epfl.balelecbud.util.database.DatabaseWrapper;
+import ch.epfl.balelecbud.util.database.MyQuery;
+
+import static ch.epfl.balelecbud.BalelecbudApplication.getAppDatabaseWrapper;
+
+public class EmergencyNumbersActivity extends BasicActivity {
+    public static final int PERMISSION_TO_CALL_CODE = 991;
+    private boolean callPermissionGranted;
+    private ListView numbersListView;
+    private ArrayAdapter arrayAdapter;
+    private Map<String, String> repertoryMap = new HashMap<String, String>();
+    List<String> numberList = new ArrayList<>(Arrays.asList(repertoryMap.keySet().toArray(new String[0])));
+
+
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        BalelecbudApplication.getAppDatabaseWrapper();
+
+        setContentView(R.layout.activity_emergency_numbers);
+        configureToolBar(R.id.emergency_numbers_activity_toolbar);
+        configureDrawerLayout(R.id.emergency_numbers_activity_drawer_layout);
+        configureNavigationView(R.id.emergency_numbers_activity_nav_view);
+
+
+        numbersListView = findViewById(R.id.numbersListView);
+        arrayAdapter = new ArrayAdapter(
+                this,
+                android.R.layout.simple_list_item_1,
+                numberList);
+
+        numbersListView.setAdapter(arrayAdapter);
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        BalelecbudApplication.getAppDatabaseWrapper().query( new MyQuery(DatabaseWrapper. EMERGENCY_NUMBER_PATH, new LinkedList<>()), EmergencyNumber.class).whenComplete((res, err) -> {
+            for (EmergencyNumber number : res) {
+                repertoryMap.put(number.getName(), number.getNumber());
+                Log.d("numer", number.getName());
+            }
+            numberList.clear();
+            numberList.addAll(new ArrayList<>(Arrays.asList(repertoryMap.keySet().toArray(new String[0]))));
```

Comment:
You could update the list instead of clearing it and re-adding all elements

Reply:
Secounded